<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Visualizer3DMapperClass</class>
 <widget class="QWidget" name="Visualizer3DMapperClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>419</width>
    <height>703</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Lidar Image Processing</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="gbFrameControl">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>Image Selection</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="wFrameControl" native="true">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Image</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="hsImageNb">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lImageName">
             <property name="text">
              <string>&quot;&quot;</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="pbProcessImage">
           <property name="text">
            <string>Process Image</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="pbProcessContinuous">
             <property name="text">
              <string>Read Continuously</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>sleep:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dsbSleep">
             <property name="suffix">
              <string> s</string>
             </property>
             <property name="minimum">
              <double>0.050000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.500000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="pbProcessNumber">
             <property name="text">
              <string>Read Certain Number</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sbReadNumber">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100000</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QPushButton" name="pbExport">
             <property name="text">
              <string>Export</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbExport3D">
             <property name="text">
              <string>Exp3D</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbImport">
             <property name="text">
              <string>Import</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbImport3D">
             <property name="text">
              <string>Imp3D</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbExportTraj">
             <property name="text">
              <string>Exp Traj</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbImportTraj">
             <property name="text">
              <string>Imp Traj</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QPushButton" name="pbSaveMap">
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbLoadMap">
             <property name="text">
              <string>Load</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbReset">
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gbFrameProcessing">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>Processing</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="wFrameProcessing" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QPushButton" name="pbAddRegisteredScan">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Add To Map</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbUnwarp">
             <property name="text">
              <string>Unwarp</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbAdept">
             <property name="text">
              <string>Adept</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbGPS">
             <property name="text">
              <string>GPS</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbRemOld">
             <property name="text">
              <string>-Old</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbRemFar">
             <property name="text">
              <string>-Far</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <widget class="QPushButton" name="pbRegister">
             <property name="text">
              <string>Register</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Subsampling</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="hsSubsampling">
             <property name="minimum">
              <number>50</number>
             </property>
             <property name="maximum">
              <number>50000</number>
             </property>
             <property name="singleStep">
              <number>10</number>
             </property>
             <property name="pageStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>2000</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lSubsamplingVal">
             <property name="text">
              <string>2000</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <widget class="QPushButton" name="pbFilterMap_init">
             <property name="text">
              <string>Filter Init/Reset</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbFilterMap_step">
             <property name="text">
              <string>Iterate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbFilterMap_finish">
             <property name="text">
              <string>Finish</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QPushButton" name="pbReg1">
             <property name="text">
              <string>Reg Init/Reset</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbReg2">
             <property name="text">
              <string>Iterate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbReg3">
             <property name="text">
              <string>Finish</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="pbReadImage">
           <property name="text">
            <string>Read Image</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pbFilterMap">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Filter Map</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gbVisualization">
     <property name="title">
      <string>Visualization</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="wVisualization" native="true">
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0" colspan="2">
          <widget class="QCheckBox" name="cbVisuEstTraject">
           <property name="text">
            <string>Estimated Trajectory</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QCheckBox" name="cbVisuGPSTraject">
           <property name="text">
            <string>GPS Trajectory</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QCheckBox" name="cbVisuMap">
           <property name="text">
            <string>Map, color by :</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="4" column="0" colspan="2">
          <widget class="QRadioButton" name="rbMapHeight">
           <property name="text">
            <string>Height</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="6" column="0" colspan="2">
          <widget class="QRadioButton" name="rbMapNormal">
           <property name="text">
            <string>Normal</string>
           </property>
          </widget>
         </item>
         <item row="6" column="2">
          <widget class="QRadioButton" name="rbMapNConf">
           <property name="text">
            <string>Normal-Confidence</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0" colspan="2">
          <widget class="QCheckBox" name="cbOrigMap">
           <property name="text">
            <string>Original Map</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0" colspan="2">
          <widget class="QCheckBox" name="cbVisuCurrPoints">
           <property name="text">
            <string>Current Points</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="8" column="2">
          <widget class="QCheckBox" name="cbVisuCurrNormals">
           <property name="text">
            <string>Current Normals</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0" colspan="2">
          <widget class="QCheckBox" name="cbVisuLastPoints">
           <property name="text">
            <string>Last Points</string>
           </property>
          </widget>
         </item>
         <item row="9" column="2">
          <widget class="QCheckBox" name="cbVisuSubPoints">
           <property name="text">
            <string>Subsampled Points</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="10" column="2">
          <widget class="QCheckBox" name="cbFiltering">
           <property name="text">
            <string>Filtering</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>min Hit Count:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sbMinHitCnt"/>
           </item>
          </layout>
         </item>
         <item row="7" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QCheckBox" name="cbAdaption">
             <property name="text">
              <string>Adaption</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sbFilterCol"/>
           </item>
           <item>
            <widget class="QSpinBox" name="sbFilterRow"/>
           </item>
          </layout>
         </item>
         <item row="2" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="pbPtPlus">
             <property name="text">
              <string>+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbPtMinus">
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="0">
          <widget class="QRadioButton" name="rbMapHeightSlices">
           <property name="text">
            <string>Height-Slices</string>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QRadioButton" name="rbMapDist">
           <property name="text">
            <string>ScannerDist</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QRadioButton" name="rbMapIntensity">
           <property name="text">
            <string>Intensity</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>gbVisualization</sender>
   <signal>toggled(bool)</signal>
   <receiver>wVisualization</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>549</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbFrameProcessing</sender>
   <signal>toggled(bool)</signal>
   <receiver>wFrameProcessing</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbFrameControl</sender>
   <signal>toggled(bool)</signal>
   <receiver>wFrameControl</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
